### PR TITLE
Add 'pipeline' parameter and schematized format to export endpoint

### PR DESCRIPTION
**Pull request by lava**
Add the ability to pass an ad-hoc pipeline definition to the 'export' endpoint.

### CODE REVIEW

File: libvast/builtins/endpoints/export.cpp [hunk 1]
```diff
@@ -170,41 +433,44 @@ export_helper(export_helper_actor::stateful_pointer<export_helper_state> self,
...
-      else
-        writer.write(truncate(slice, remaining));
-      self->state.events_ += std::min<size_t>(slice.rows(), remaining);
-      self->state.stringified_events_
-        += static_cast<std::stringstream&>(writer.out()).str();
+    [self](vast::table_slice& slice) {
+      std::vector<table_slice> slices;
+      if (self->state.pipeline_) {
+        self->state.pipeline_->add(std::move(slice));
+        auto transformed = self->state.pipeline_->finish();
```
Comment (lava) [inline libvast/builtins/endpoints/export.cpp:440]: Only apply the pipeline after all events up to `limit` have been received.

File: libvast/builtins/endpoints/export.cpp [hunk 2]
```diff
@@ -130,17 +249,149 @@ using export_multiplexer_actor = system::typed_actor_fwd<>
...
+/// Format a set of table slices like this:
+// clang-format off
+//
+// {
+//   "version": "v2.4-rc2",
+//   "num_events": 3,
+//   "events": [
+//      {
+//        "name": "zeek.conn",
+//        "schema": [{"name": "_path", "type": "string"}, {"name": "uid", "type": "string"}, ...]
```
Comment (lava) [inline libvast/builtins/endpoints/export.cpp:324]: Update comment to new schema formatting

File: libvast/builtins/endpoints/export.cpp [hunk 3]
```diff
@@ -243,17 +497,68 @@ export_multiplexer_actor::behavior_type export_multiplexer(
...
+        params.format_opts.omit_nulls = caf::get<bool>(param);
+      }
+      if (rq.params.contains("numeric-durations")) {
+        auto& param = rq.params.at("numeric-durations");
+        VAST_ASSERT(caf::holds_alternative<bool>(param));
+        params.format_opts.numeric_durations = caf::get<bool>(param);
+      }
+      auto pipeline_executor = std::optional<vast::pipeline_executor>{};
+      if (rq.params.contains("pipeline")) {
+        // {"steps": []}
```
Comment (lava) [inline libvast/builtins/endpoints/export.cpp:528]: Remove comment

**Review by tobim — APPROVED** [review]
We reviewed this in a pairing session.

**Pull request merged**